Source/SwiftLintFramework/Protocols/Rule.swift

```diff
@@ -47,6 +47,61 @@ public extension CorrectableRule {
     }
 }
 
+public protocol SubstitutionCorrectableRule: CorrectableRule {
+    func violationRanges(in file: File) -> [NSRange]
+    func substitution(for violationRange: NSRange, in file: File) -> (NSRange, String)
+}
+
+public extension SubstitutionCorrectableRule {
+    func correct(file: File) -> [Correction] {
+        let violatingRanges = file.ruleEnabled(violatingRanges: violationRanges(in: file), for: self)
+        guard !violatingRanges.isEmpty else { return [] }
+
+        let description = type(of: self).description
+        var corrections = [Correction]()
+        var contents = file.contents
+        for range in violatingRanges {
```

Comment:
I wonder if we should sort here just to be sure it's always correct

Reply:
Yes we should always sort. Applying corrections, especially multiple ones in a file, is so rare in practice compared to the amount of code being linted that a slight (probably immeasurable) perf hit is worth the wins in correctness, maintainability and readability.

Reply:
Would be a good move to move the sorting here, since you could be overwritting `violationRanges` function, and forgot the sorting. Place for great frustration upon trying to figure out why your corrections are all messed up 🤯